### PR TITLE
[CPU] Restore legacy SetBlob and GetBlob to API 1.0 version

### PR DESCRIPTION
**Pull request by mandrono**
### Details:
 - *item1*
 - *...*

### Tickets:
 - *78879*

### CODE REVIEW

**Comment by dmitry-gorokhov** [comment]
@maxnick please review the patch

File: src/bindings/python/tests_compatibility/__init__.py [hunk 1]
```diff
@@ -138,3 +138,5 @@ def xfail_test(reason="Mark the test as expected to fail", strict=True):
 xfail_issue_63643 = xfail_test(reason="RuntimeError: Unsupported operation of type: Convolution name")
 
 xfail_issue_77668 = xfail_test(reason="Accuracy issue related to Gather-8.")
+
+xfail_issue_TBD = xfail_test(reason="Cannot get dims for non static shape")
```
Comment (mandrono) [inline src/bindings/python/tests_compatibility/__init__.py:142]:
```suggestion
xfail_issue_78741 = xfail_test(reason="Cannot get dims for non static shape")
```

File: src/plugins/intel_cpu/src/mkldnn_infer_request.cpp [hunk 2]
```diff
@@ -467,36 +443,31 @@ void MKLDNNPlugin::MKLDNNLegacyInferRequest::SetBlob(const std::string& name, co
...
+        if (data->getTensorDesc().getLayout() != InferenceEngine::Layout::ANY && foundOutput->getTensorDesc().getLayout() != InferenceEngine::Layout::ANY &&
+            foundOutput->getTensorDesc().getBlockingDesc() != data->getTensorDesc().getBlockingDesc()) {
                 IE_THROW(ParameterMismatch) << "Failed to set output blob. Blocking descriptor mismatch.";
         }
 
-        const auto &desc = graph->getOutputNodeByName(name)->getParentEdgesAtPort(0)[0]->getMemory().getDesc();
-        if (!isDynamic && blobDesc == MemoryDescUtils::convertToTensorDesc(desc) && !graph->getProperty().batchLimit) {
+        auto pBlob = MemoryDescUtils::interpretAsBlob(graph->getOutputNodeByName(name)->getParentEdgesAtPort(0)[0]->getMemory());
+        if (!pBlob)
+            IE_THROW() << "MKLDNN graph doesn't contain output node with name: " << name;
```
Comment (maxnick) [inline src/plugins/intel_cpu/src/mkldnn_infer_request.cpp:467]: The same thing with the error message

File: src/plugins/intel_cpu/src/mkldnn_infer_request.cpp [hunk 3]
```diff
@@ -525,25 +495,30 @@ InferenceEngine::Blob::Ptr MKLDNNPlugin::MKLDNNLegacyInferRequest::GetBlob(const
         }
 
         if (_inputs.find(name) == _inputs.end()) {
+            auto pBlob = MemoryDescUtils::interpretAsBlob(graph->getInputNodeByName(name)->getChildEdgesAtPort(0)[0]->getMemory());
+            if (!pBlob) {
+                IE_THROW() << "MKLDNN graph doesn't contain input node with name: " << name;
+            }
```
Comment (maxnick) [inline src/plugins/intel_cpu/src/mkldnn_infer_request.cpp:501]: The same question about the error message

File: src/plugins/intel_cpu/src/mkldnn_infer_request.cpp [hunk 4]
```diff
@@ -564,67 +539,52 @@ InferenceEngine::Blob::Ptr MKLDNNPlugin::MKLDNNLegacyInferRequest::GetBlob(const
...
-                            if (expectedTensorDesc.getBlockingDesc() != tensorDesc.getBlockingDesc())
-                                IE_THROW(ParameterMismatch) << "Network input and output use the same name: " << name
-                                                            << " but expect blobs with different blocking descriptors.";
-                        }
-                    }
+        if (_outputs.find(name) == _outputs.end()) {
+            auto pBlob = MemoryDescUtils::interpretAsBlob(graph->getOutputNodeByName(name)->getParentEdgesAtPort(0)[0]->getMemory());
+            if (!pBlob) {
+                IE_THROW() << "MKLDNN graph doesn't contain output node with name: " << name;
+            }
```
Comment (maxnick) [inline src/plugins/intel_cpu/src/mkldnn_infer_request.cpp:546]: The same thing about the error message

**Review by maxnick — APPROVED** [review]

**Review by dmitry-gorokhov — APPROVED** [review]

**Pull request merged**